packages/jest-cli/src/cli/index.js

```diff
@@ -8,17 +8,92 @@
  * @flow
  */
 
-import type {Path} from 'types/Config';
+import type {AggregatedResult} from 'types/TestResult';
 import type {Argv} from 'types/Argv';
+import type {GlobalConfig, Path, ProjectConfig} from 'types/Config';
 
-import {validateCLIOptions} from 'jest-util';
-import yargs from 'yargs';
+import {
+  Console,
+  clearLine,
+  createDirectory,
+  validateCLIOptions,
+} from 'jest-util';
+import {readConfig} from 'jest-config';
+import {version as VERSION} from '../../package.json';
 import args from './args';
+import chalk from 'chalk';
+import createContext from '../lib/createContext';
 import getJest from './getJest';
-import runCLI from './runCLI';
+import getMaxWorkers from '../lib/getMaxWorkers';
+import handleDeprecationWarnings from '../lib/handleDeprecationWarnings';
+import logDebugMessages from '../lib/logDebugMessages';
+import preRunMessage from '../preRunMessage';
+import runJest from '../runJest';
+import Runtime from 'jest-runtime';
+import TestWatcher from '../TestWatcher';
+import watch from '../watch';
+import yargs from 'yargs';
+
+function run(maybeArgv?: Argv, project?: Path) {
```

Comment:
`run` and `runCLI` are two main functions that get exported and both of them are just a few lines now (all data massaging/parsing is modularized and extracted into separate functions).

ideally i would name `run` as `findJestAndRun`, but those are part of public interface